resolver/resolver_test.go

```diff
@@ -309,127 +84,168 @@ func TestNonMesosHandler(t *testing.T) {
+					net.ParseIP("1.1.1.1"))),
+				nth(1, answers, a(
+					rrheader("google.com.", dns.TypeA, 60),
+					net.ParseIP("2.2.2.2")))),
+		},
+	} {
+		var rw responseRecorder
+		tt.handle(&rw, &tt.query)
+		errs := tt.assert(rw.msg)
+		if len(errs) == 0 {
+			continue
+		}
+		t.Logf("Test #%d\n%v\n", i, pretty.Sprint(tt.query))
+		for _, err := range errs {
+			t.Error(err)
+		}
 	}
 }
 
 func TestHTTP(t *testing.T) {
-
 	// setup DNS server (just http)
-	res, err := fakeDNS(8053)
-	if err != nil {
-		t.Error(err)
-	}
+	res := fakeDNS(t)
 	res.version = "0.1.1"
 
 	res.configureHTTP()
 	ts := httptest.NewServer(http.DefaultServeMux)
 	defer ts.Close()
 
-	// test /v1/version
-	r1, err := http.Get(ts.URL + "/v1/version")
-	if err != nil {
-		t.Error(err)
-	}
-	g1, err := ioutil.ReadAll(r1.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got1 map[string]interface{}
-	err = json.Unmarshal(g1, &got1)
-	correct1 := map[string]interface{}{"Service": "Mesos-DNS", "Version": "0.1.1", "URL": "https://github.com/mesosphere/mesos-dns"}
-	eq1 := reflect.DeepEqual(got1, correct1)
-	if !eq1 {
-		t.Error("HTTP version API failure")
-	}
-
-	// test /v1/config
-	r2, err := http.Get(ts.URL + "/v1/config")
-	if err != nil {
-		t.Error(err)
-	}
-	g2, err := ioutil.ReadAll(r2.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got2 records.Config
-	err = json.Unmarshal(g2, &got2)
-	eq2 := reflect.DeepEqual(got2, res.config)
-	if !eq2 {
-		t.Error("HTTP config API failure")
-	}
-
-	// test /v1/services -- existing record
-	r3, err := http.Get(ts.URL + "/v1/services/_leader._tcp.mesos.")
-	if err != nil {
-		t.Error(err)
-	}
-	g3, err := ioutil.ReadAll(r3.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got3 []map[string]interface{}
-	err = json.Unmarshal(g3, &got3)
-	correct3 := []map[string]interface{}{{"host": "leader.mesos.", "port": "5050", "service": "_leader._tcp.mesos.", "ip": "1.2.3.4"}}
-	eq3 := reflect.DeepEqual(got3, correct3)
-	if !eq3 {
-		t.Error("HTTP services API failure")
-	}
-
-	// test /v1/services -- non existing record
-	r4, err := http.Get(ts.URL + "/v1/services/_myservice._tcp.mesos.")
-	if err != nil {
-		t.Error(err)
-	}
-	g4, err := ioutil.ReadAll(r4.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got4 []map[string]interface{}
-	err = json.Unmarshal(g4, &got4)
-	correct4 := []map[string]interface{}{{"host": "", "port": "", "service": "", "ip": ""}}
-	eq4 := reflect.DeepEqual(got4, correct4)
-	if !eq4 {
-		t.Error("HTTP services API failure")
-	}
-
-	// test /v1/host -- existing record
-	r5, err := http.Get(ts.URL + "/v1/hosts/leader.mesos")
-	if err != nil {
-		t.Error(err)
-	}
-	g5, err := ioutil.ReadAll(r5.Body)
-	if err != nil {
-		t.Error(err)
-	}
-	var got5 []map[string]interface{}
-	err = json.Unmarshal(g5, &got5)
-	correct5 := []map[string]interface{}{{"host": "leader.mesos.", "ip": "1.2.3.4"}}
-	eq5 := reflect.DeepEqual(got5, correct5)
-	if !eq5 {
-		t.Error("HTTP hosts API failure")
+	for _, tt := range []struct {
```

Comment:
You can't use `t` since it's the test function argument. I'd stick to the convention.